naga/xtask/src/main.rs

```diff
@@ -25,6 +26,8 @@ fn main() -> ExitCode {
         .format_indent(Some(0))
         .init();
 
+    jobserver::init();
+
```

Comment:
The [jobserver docs](https://docs.rs/jobserver/latest/jobserver/struct.Client.html#method.from_env_ext) say:

> It’s recommended to call \[the jobserver initialization function] very early in the lifetime of a program before any other file descriptors are opened. That way you can make sure to take ownership properly of the file descriptors passed down, if any.

That's why we have the call to `init` at the top of main.

The jobserver may take ownership of file descriptors passed down from a parent process, so you're only supposed to create one `Client` per process. For this, the `static` approach makes the most sense.

The jobserver is simply a pipe (on Unix) or a semaphore (on Windows). It's not expensive.

There are certainly plenty of other ways to have a thread pool of an appropriate size run the jobs, many of which would be more Rustic than jobserver, which has the feel of a 1980's Unix hack (which is what it is). The advantage of jobserver is that it will help Naga's `cargo xtask` cooperate properly when used as part of some larger build process, coordinated by `make` or another tool that uses the jobserver protocol. I don't know of anyone who's doing that right now, but when the question is "I want to run as many automation jobs concurrently as possible", it seems like using the defacto standard for whole-task coordination is a good thing.

Reply:
If there's a suspicion that `naga xtask` may need to integrate with a concrete job server later, then I think this is fine. Doing so speculatively makes me uncomfortable (because simpler code plz; this has been your schtick for perf.), but I'd consider it your decision to make.

I consider this conversation resolved, since I feel I've delivered the point I wanted to make.